athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/common/StorageLocation.java

```diff
@@ -0,0 +1,68 @@
+/*-
+ * #%L
+ * athena-gcs
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs.common;
+
+import static java.util.Objects.requireNonNull;
+
+public class StorageLocation
+{
+    private final String bucketName;
+    private String location;
+
+    public StorageLocation(String bucketName, String location)
+    {
+        this.bucketName = requireNonNull(bucketName, "Bucket name was null");
+        this.location = requireNonNull(location, "Location name was null");
+        if (!this.location.endsWith("/")) {
+            this.location += "/";
+        }
+    }
+
+    public String getBucketName()
+    {
+        return bucketName;
+    }
```

Comment:
This class doesn't seem like it needs to exist.
You can just use `java.net.URI` instead.

Here's an example of how that class is used:
```
jshell> var uri  = new URI("gcs://someasdfbucket/asdfasdfasdf/dfsaffdsg/");
uri ==> gcs://someasdfbucket/asdfasdfasdf/dfsaffdsg/

jshell> uri.getScheme();
$10 ==> "gcs"

jshell> uri.getAuthority();
$11 ==> "someasdfbucket"

jshell> uri.getPath();
$12 ==> "/asdfasdfasdf/dfsaffdsg/"
```